opsmngr/logs.go

```diff
@@ -0,0 +1,285 @@
+)
+
+const (
+	logsBasePath = "groups/%s/logCollectionJobs"
+)
+
+// LogCollectionService is an interface for interfacing with the Log Collection Jobs
+// endpoints of the MongoDB Ops Manager API.
+// See more: https://docs.opsmanager.mongodb.com/current/reference/api/log-collection/
+type LogCollectionService interface {
+	List(context.Context, string, *LogListOptions) (*LogCollectionJobs, *atlas.Response, error)
+	Get(context.Context, string, string, *LogListOptions) (*LogCollectionJob, *atlas.Response, error)
+	Retry(context.Context, string, string) (*atlas.Response, error)
+	Create(context.Context, string, *LogCollectionJob) (*LogCollectionJob, *atlas.Response, error)
+	Extend(context.Context, string, string, *LogCollectionJob) (*atlas.Response, error)
+	Delete(context.Context, string, string) (*atlas.Response, error)
+}
+
+// LogsService is an interface for interfacing with the Log Collection Jobs
+// endpoints of the MongoDB Ops Manager API.
+// See more: https://docs.opsmanager.mongodb.com/current/reference/api/log-collection/
+type LogsService interface {
+	Download(context.Context, string, string, io.Writer) (*atlas.Response, error)
+}
+
+// LogCollectionServiceOp handles communication with the Log Collection Jobs related methods of the
+// MongoDB Ops Manager API
+type LogCollectionServiceOp struct {
+	Client atlas.RequestDoer
+}
+
+// LogCollectionServiceOp handles communication with the Log Collection Jobs download method of the
+// MongoDB Ops Manager API
+type LogsServiceOp struct {
+	Client atlas.GZipRequestDoer
+}
+
+var _ LogCollectionService = &LogCollectionServiceOp{}
+var _ LogsService = &LogsServiceOp{}
+
+// LogCollectionJob represents a Log Collection Job in the MongoDB Ops Manager API.
+type LogCollectionJob struct {
+	ID                         string       `json:"id,omitempty"`
+	GroupID                    string       `json:"groupId,omitempty"`
+	UserID                     string       `json:"userId,omitempty"`
+	CreationDate               string       `json:"creationDate,omitempty"`
+	ExpirationDate             string       `json:"expirationDate,omitempty"`
+	Status                     string       `json:"status,omitempty"`
+	ResourceType               string       `json:"resourceType,omitempty"`
+	ResourceName               string       `json:"resourceName,omitempty"`
+	RootResourceName           string       `json:"rootResourceName,omitempty"`
+	RootResourceType           string       `json:"rootResourceType,omitempty"`
+	URL                        string       `json:"downloadUrl,omitempty"`
+	Redacted                   *bool        `json:"redacted"`
+	LogTypes                   []string     `json:"logTypes,omitempty"`
+	SizeRequestedPerFileBytes  int64        `json:"sizeRequestedPerFileBytes,omitempty"`
+	UncompressedDiskSpaceBytes int64        `json:"uncompressedSizeTotalBytes,omitempty"`
+	ChildJob                   []*ChildJobs `json:"childJobs,omitempty"` //included if verbose is true
```

Comment:
This is the other way around of what I said before
```suggestion
	ChildJobs                   []*ChildJob `json:"childJobs,omitempty"` //included if verbose is true
```

`LogCollectionJob.ChildJobs` is a collection (slice/array) of `type ChildJob struct` where `type ChildJob struct` is just one
(1) Job instance. Like I said before is a good practice to structure your code this way makes it easier to reason about between one and multiple elements of the same type